exchanges/bitfinex/bitfinex_websocket.go

```diff
@@ -1073,6 +494,617 @@ func (b *Bitfinex) wsHandleData(respRaw []byte) error {
 	return nil
 }
 
+func (b *Bitfinex) handleWSChannelUpdate(c *stream.ChannelSubscription, d []interface{}) error {
+	switch c.Channel {
+	case wsBook:
+		return b.handleWSBookUpdate(c, d)
+	case wsCandles:
+		return b.handleWSCandleUpdate(c, d)
+	case wsTicker:
+		return b.handleWSTickerUpdate(c, d)
+	case wsTrades:
+		return b.handleWSTradesUpdate(c, d)
+	}
+
+	return fmt.Errorf("%s unhandled channel update: %s", b.Name, c.Channel)
+}
+
+func (b *Bitfinex) handleWSBookUpdate(c *stream.ChannelSubscription, d []interface{}) error {
+	var newOrderbook []WebsocketBook
+	obSnapBundle, ok := d[1].([]interface{})
+	if !ok {
+		return errors.New("orderbook interface cast failed")
```

Comment:
Could you please standardise type assertion error with `common.GetTypeAssertError` across just your changes.

Reply:
Yep.

Reply:
FWIW, I just moved these around.

Reply:
Actually, this is a huge request, and not something I introduced.
I've reached my yak limit on this. I'll add it for a separate task.

Reply:
Also: I think we'd use one of:
* [array to struct](https://eagain.net/articles/go-json-array-to-struct/)
* And there's a note from @thrasher- to consider b2dfef87989d1ecbe35822e0cd973bb47ada150a